### DISCUSSION TITLE
Several LSP services for a language

### DISCUSSION THREAD

**Discussion by romannikov** [zed-industries/zed#18583] [Feature Requests]
Is it possible to have several LSP services per a file/language?
All examples in the exceptions repo are one LSP for one language, but can I extend some LSP by adding an extension with my own LSP?

Many thanks!

**Comment by bajrangCoder** [comment]
Yes , you can create your own LSP extension and add the language code in that extension's `extension.toml`, some thing like this:
https://github.com/zed-industries/zed/blob/563a1dcbab67b58a6c9cf10fe4dea35d862376d1/extensions/emmet/extension.toml#L12

Then from your settings file you can enable/disable language servers, which you will need.
Check this: https://zed.dev/docs/configuring-languages#choosing-language-servers

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
